### ISSUE TITLE
The Volatile GPU-Util is zero when training. how to speed up?

### ISSUE THREAD

**Issue by zimurui** [pierluigiferrari/ssd_keras#216]
Hello, thank you for your share of ssd code in keras, backend tensorflow.
I run the ssd300_train.py with VOC07-12 dataset with three GTX1080, but the each GPU-Util is always shown  zero, as shown bellow.
I was firstly wondering the data input pipiline is more slower than the training operation on GPU, but the keras  says model.fit_generator,  "The generator is run in parallel to the model, for efficiency. For instance, this allows you to do real-time data augmentation on images on CPU in parallel to training your model on GPU."
So i am confused that what reason for the zero GPU use ratio?

![11](https://user-images.githubusercontent.com/32056332/51252480-e52ae480-19d6-11e9-9a9a-142861e126cd.jpg)

**Comment by pierluigiferrari** [comment]
This is because the preprocessing of a batch (loading, data augmentation, and encoding of the labels) on the CPU takes longer than a forward and backward pass on the GPU, so the preparation of every batch takes longer than the respective training iteration.

The solution would be to build a significantly faster input pipeline. One of the best ways to speed up the input pipeline would be to implement it entirely in TensorFlow. I'm working on a new input pipeline based on TFRecords and contained entirely in TensorFlow, but I can't say when it will be ready. The preprocessing of SSD is complex and it requires careful testing to make sure it's correct (especially the matching / label encoding).
